### PR TITLE
Added Open ID Connect (oidc) compatibility

### PR DESCRIPTION
**Pull request by dan-mcm**
Users can access the oidc feature with the following command:

`zign oidc -c $clientId`

Default auth and token url set to Team Wolves identity provider (Dex). Users can specify commands with the following:

-n (token name)
-a (authroize-url)
-t (token-url)
-c (client-id)
-p (business-partner-id)
-r (refresh)

The command `zign oidc -h` will provide a full list of commands available.

### CODE REVIEW

**Comment by hjacobs** [comment]
Can you clarify the intentions of this PR? What is the purpose? Your use case (own identity provider) sounds a bit disconnected / out of scope.

**Comment by grahamar** [comment]
Hey @hjacobs [OpenID Connect](http://openid.net/connect/) is a standard, not specific to us. We just happen to have an OpenID Connect compliant identity provider we need something like Zign for.

This is a relevant supplement to the OAuth functionality currently in zign as OAuth 2.0 capabilities are integrated with the OIDC protocol itself. We'd like to commit this back as it would be useful to the open source community to have a tool for generating access tokens for an OIDC compliant identity provider.

PlatformIAM is close to being an OpenID Connect provider and if it's ever decided to be a fully compliant OIDC provider, this logic will be needed.

File: zign/cli_zign.py [hunk 1]
```diff
@@ -51,6 +53,27 @@ def token(obj, scope, url, realm, name, user, password, insecure, refresh):
...
+@click.option('-a', '--authorize-url', help='OAuth 2 Authorization Endpoint URL to generate access token',
+              metavar='AUTH_URL', default='https://identity.wolves.zalan.do/auth')
+@click.option('-t', '--token-url', help='OAuth 2 Token URL to generate access token using a refresh token',
+              metavar='TOKEN_URL', default='https://identity.wolves.zalan.do/token')
+@click.option('-c', '--client-id', help='Client ID to use', metavar='CLIENT_ID')
+@click.option('-p', '--business-partner-id', help='Business Partner ID to use', metavar='PARTNER_ID')
+@click.option('-r', '--refresh', help='Force refresh of the access token', is_flag=True, default=False)
+
+def fcptoken(name, authorize_url, token_url, client_id, business_partner_id, refresh):
+    '''Create a new Platform IAM token or use an existing one.'''
```
Comment (grahamar) [inline zign/cli_zign.py]: This method needs to be renamed to oidc, not fcp... And documentation updated accordingly.

File: zign/cli_zign.py [hunk 2]
```diff
@@ -51,6 +53,27 @@ def token(obj, scope, url, realm, name, user, password, insecure, refresh):
     access_token = get_token(name, scope)
     print(access_token)
 
+# FCP specific zign token for generating OIDC token
+@cli_zign.command('oidc')
+@click.option('-n', '--name', help='Custom token name (will be stored)', metavar='TOKEN_NAME', default='fcp_app_token')
```
Comment (grahamar) [inline zign/cli_zign.py]: Replace `fcp_app_token` with `oidc_id_token`

File: zign/cli_zign.py [hunk 3]
```diff
@@ -51,6 +53,27 @@ def token(obj, scope, url, realm, name, user, password, insecure, refresh):
     access_token = get_token(name, scope)
     print(access_token)
 
+# FCP specific zign token for generating OIDC token
```
Comment (grahamar) [inline zign/cli_zign.py]: Update comment, this is not FCP specific

File: zign/oidc_api.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
+import logging
```
Comment (grahamar) [inline zign/oidc_api.py:1]: It looks like a lot of this is duplicated code from the existing `api.py` file, @daniel40392 can you refactor?

**Pull request closed without merging**